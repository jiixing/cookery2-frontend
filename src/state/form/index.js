import * as constants from './constants'

export { constants }
export { default as reducer } from './reducers'

export { modelName } from './utils'
