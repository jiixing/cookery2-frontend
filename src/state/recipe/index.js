import * as constants from './constants'
import * as actions from './actions'

export { actions, constants }
export { default as recipeSelector } from './selectors'
export { default as reducer } from './reducers'
