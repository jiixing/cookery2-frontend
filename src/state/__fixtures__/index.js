import recipes from './recipes'
import form from './form'

import * as rest from './rest'

const recipe = () => rest.recipe
const state = { recipes: recipes(), recipe: recipe(), form: form() }

export { recipes, recipe, rest }
export default () => state
