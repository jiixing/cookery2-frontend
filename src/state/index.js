import { combineReducers } from 'redux'

import {
  constants as recipesConstants,
  reducer as recipesReducer,
} from './recipes'
import {
  constants as recipeConstants,
  reducer as recipeReducer,
} from './recipe'
import { constants as formConstants, reducer as formReducer } from './form'

const rootReducer = combineReducers({
  [recipesConstants.namespace]: recipesReducer,
  [recipeConstants.namespace]: recipeReducer,
  [formConstants.namespace]: formReducer,
})

export default rootReducer
