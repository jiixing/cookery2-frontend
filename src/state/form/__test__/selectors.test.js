import state from '../../__fixtures__'
import {
  formSelector,
  recipeSelector,
  fieldSelectorBuilder,
} from '../selectors'

describe('selectors', () => {
  describe('formSelector', () => {
    it('selects the form', () => {
      expect(formSelector(state())).toMatchSnapshot()
    })
  })

  describe('recipeSelector', () => {
    it('selects the form', () => {
      expect(recipeSelector(state())).toMatchSnapshot()
    })
  })

  describe('fieldSelectorBuilder', () => {
    it('selects a concrete field', () => {
      expect(fieldSelectorBuilder('name')(state())).toMatchSnapshot()
    })
  })
})
