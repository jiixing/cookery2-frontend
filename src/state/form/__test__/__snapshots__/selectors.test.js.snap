// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectors fieldSelectorBuilder selects a concrete field 1`] = `
Object {
  "errors": Object {},
  "focus": false,
  "initialValue": "",
  "intents": Array [],
  "model": "form.recipe.name",
  "pending": false,
  "pristine": true,
  "retouched": false,
  "submitFailed": false,
  "submitted": false,
  "touched": false,
  "valid": true,
  "validated": false,
  "validating": false,
  "validity": Object {},
  "value": "",
}
`;

exports[`selectors formSelector selects the form 1`] = `
Object {
  "errors": Object {},
  "focus": false,
  "initialValue": Object {
    "difficulty": "easy",
    "duration": 0,
    "name": "",
    "servings": 1,
  },
  "intents": Array [],
  "model": "form.recipe",
  "pending": false,
  "pristine": true,
  "retouched": false,
  "submitFailed": false,
  "submitted": false,
  "touched": false,
  "valid": true,
  "validated": false,
  "validating": false,
  "validity": Object {},
  "value": Object {
    "difficulty": "easy",
    "duration": 0,
    "name": "",
    "servings": 1,
  },
}
`;

exports[`selectors recipeSelector selects the form 1`] = `
Object {
  "difficulty": "easy",
  "duration": 0,
  "name": "",
  "servings": 1,
}
`;
